crates/services/producer/src/ports.rs

```diff
@@ -34,6 +39,20 @@ pub trait BlockProducerDatabase: Send + Sync {
     /// Gets the committed block at the `height`.
     fn get_block(&self, height: &BlockHeight) -> StorageResult<Cow<CompressedBlock>>;
 
+    /// Gets the full committed block at the `height`.
+    fn get_full_block(&self, height: &BlockHeight) -> StorageResult<Block> {
+        let block = self.get_block(height)?;
+        let transactions = block
+            .transactions()
+            .iter()
+            .map(|id| self.get_transaction(id).map(|tx| tx.into_owned()))
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok(block.into_owned().uncompress(transactions))
+    }
```

Comment:
I meant we can move `get_full_block` implementation to the `fuel-core`. Later, when we optimize the getting of the full block, we will avoid iteration over transactions. Plus it will be clear from the business perspective

Reply:
I still don't understand where it would live to be part of `fuel-core`.

Reply:
In the implementation of the `BlockProducerDatabase` port in `fuel-core`.

<img width="1330" alt="image" src="https://github.com/user-attachments/assets/89a3315a-2ef6-48ea-839e-8574927b5817" />

Reply:
There? Ok sure. Moved in 582a30c573.